keycloak-plugin/sasl-plugin/src/main/java/io/enmasse/keycloak/spi/K8sServiceAccountCredentialProvider.java

```diff
@@ -59,67 +59,66 @@ public boolean isConfiguredFor(RealmModel realmModel, UserModel userModel, Strin
 
     @Override
     public boolean isValid(RealmModel realmModel, UserModel userModel, CredentialInput credentialInput) {
-        JsonObject body = new JsonObject();
-
-        body.put("kind", "TokenReview");
-        body.put("apiVersion", "authentication.k8s.io/v1beta1");
-
-        JsonObject spec = new JsonObject();
-        spec.put("token", ((UserCredentialModel)credentialInput).getValue());
-        body.put("spec", spec);
+        String token = ((UserCredentialModel) credentialInput).getValue();
+        String userName = authenticateToken(token, client, httpClient);
 
-        JsonObject responseBody= doRawHttpRequest("/apis/authentication.k8s.io/v1beta1/tokenreviews", "POST", body, false);
-
-        JsonObject status = responseBody.getJsonObject("status");
-
-        boolean authenticated = false;
-        String userName = null;
-        if (status != null) {
-            Boolean auth = status.getBoolean("authenticated");
-            authenticated = auth == null ? false : auth;
-            JsonObject user = status.getJsonObject("user");
-            if (user != null) {
-                userName = user.getString("username");
-            }
-        }
-        if(!authenticated) {
+        boolean authenticated = Objects.equals(userModel.getUsername(), userName);
+        if(userName == null) {
             LOG.debug("User: " + userModel.getUsername() + " not authenticated for realm " + realmModel.getName());
-        } else if(!Objects.equals(userModel.getUsername(), userName)) {
+        } else if(!authenticated) {
             LOG.debug("Attempt to log in for user " + userModel.getUsername() + " in realm " + realmModel.getName() + " with token for " + userName);
         }
 
-        return authenticated && Objects.equals(userModel.getUsername(), userName);
+        return authenticated;
     }
 
-    private JsonObject doRawHttpRequest(String path, String method, JsonObject body, boolean errorOk) {
+    public static String authenticateToken(String token, NamespacedOpenShiftClient client, OkHttpClient httpClient) {
+        JsonObject body = new JsonObject();
+        String userName = null;
+        body.put("kind", "TokenReview");
+        body.put("apiVersion", "authentication.k8s.io/v1beta1");
```

Comment:
Note: as of 1.9/3.9 we can use 'v1', so we should change to that across the entire code base.